File: Algorithms/C++/reverse_string.cpp
#include <iostream>
#include <string>

using namespace std;

void reverse_string(string &str)
{
    int size = str.size();
    char temp;

    for (int i = 0; i < size / 2; i++)
    {
        temp = str[i];
        str[i] = str[size - i - 1];
        str[size - i - 1] = temp;
    }
}

int main()
{
    string str = "Hello World";

    reverse_string(str);
    cout << str << endl;

    return 0;
}

File: Algorithms/Javascript/reverseString.js
const reverseString = (str) => {
    return str.split('').reverse().join('');

    // return str.split('').reduce((acc, char) => char + acc, '');
    // return str.split('').reduceRight((acc, char) => acc + char, '');

    /*
    let reversed = '';
    for (let char of str) {
        reversed = char + reversed;
    }
    return reversed;
    */
}

console.log(reverseString('hello')); // olleh
